### PR TITLE
Preserve ArgoCD Notification state

### LINKED ISSUE #180: ApplicationSet conflicts with ArgoCD-Notification which cause notification spam
**ArgoCD-Notification** sets annotation to track already sent notifications:
```
apiVersion: argoproj.io/v1alpha1
kind: Application
metadata:
  annotations:
    notified.notifications.argoproj.io: >-
      {"b620d4600c771a6f4cxxxxxxx:on-deployed:[0].y7b5sbwa2Q329JYHxxxxxx-fBs:slack:slack-test":1617144614}
```
**ApplicationSet** clears external annotation on sync which leads to notification spam in the loop. 
Should be related: https://github.com/argoproj-labs/applicationset/issues/154

### PR DESCRIPTION
**Pull request by dctrwatson**
While #186 would probably be a long-term enhancement, this at least fixes #180 in the near term

### CODE REVIEW

**Review by shkrid — APPROVED** [review]

**Comment by shkrid** [comment]
I have build docker image from this PR 
`docker pull shkrid/applicationset:1.0.0-notification-fix`
and try, but issue was not resolved.

@dctrwatson

**Comment by dctrwatson** [comment]
> I have built docker image from this PR
> `docker pull shkrid/applicationset:1.0.0-notification-fix`
> and tried, but issue was not resolved.

Interesting... I'm also running an image from this PR and it is behaving as expected

The applicationset image I'm running is also based off this PR: https://github.com/argoproj-labs/argocd-notifications/pull/241

But I'm not seeing any differences to the annotation key used by the notifications controller in recent history either, so not sure :(

File: pkg/controllers/applicationset_controller.go [hunk 1]
```diff
@@ -438,7 +445,16 @@ func (r *ApplicationSetReconciler) createOrUpdateInCluster(ctx context.Context,
 		action, err := utils.CreateOrUpdate(ctx, r.Client, found, func() error {
 			// Copy only the Application/ObjectMeta fields that are significant, from the generatedApp
 			found.Spec = generatedApp.Spec
+
+			// Preserve argo cd notifications state
```
Comment (jgwest) [inline pkg/controllers/applicationset_controller.go:449]: Tiny tweak:
Can you change this line to
```
// Preserve argo cd notifications state (https://github.com/argoproj-labs/applicationset/issues/180)
```
(ie appending the issue # for some context)

Comment (dctrwatson) [reply pkg/controllers/applicationset_controller.go]: done :+1:

**Review by OmerKahani — APPROVED** [review]
Thanks!

**Review by jgwest — APPROVED** [review]
LGTM, thanks @dctrwatson

**Comment by bitgandtter** [comment]
any ETA on this to be released?

**Pull request merged**